segmentation/utils/synthetic_spatial_datagen.py

```diff
@@ -222,10 +221,163 @@ def generate_test_label_map(size_img=(1024, 1024), num_A=100, num_B=100, num_C=1
+    """
+    This function generates membrane signal for the provided cells
+
+    Args:
+        segmentation_mask (numpy): an array which contains the labeled cell regions
+        cell_centers (dict): a dictionary which contains the centers associated with each cell region
+        cell_ids (list): a list of cells we wish to generate nuclear signal for, if None assume just cell 2
+        size_img (tuple): the dimensions of the image we wish to generate
+        cell_radius (int): the radius of the entire cell, needed to do proper circle subtraction
+            for a ring-shaped membrane
+        memb_diameter (int): the diameter of the membrane ring of each cell
+
+    Returns:
+        sample_membrane_signal (numpy): an array of equal dimensions to segmentation_mask
+            which have membrane signal generated for the provided cell ids
+    """
+
+    # define the nuclear signal array
+    sample_membrane_signal = np.zeros(segmentation_mask.shape)
+
+    # default to cell 2 if cell_ids not set
+    if not cell_ids:
+        cell_ids = [2]
+
+    for cell in cell_ids:
+        center = cell_centers[cell]
+
+        # generate both the coordinates of the cell region and non-membrane region
+        # for proper circle subtraction to generate membrane
+        cell_region_x, cell_region_y = circle(center[0], center[1], cell_radius, shape=size_img)
+        non_memb_region_x, non_memb_region_y = circle(center[0], center[1], cell_radius, shape=size_img)
+
+        # perform circle subtraction
+        sample_membrane_signal[cell_region_x, cell_region_y] = 1
+        sample_membrane_signal[non_memb_region_x, non_memb_region_y] = 1
+
+        # let's keep things simple for now and not include jitter or anything
+        # that can easily be included in the next commit
+
+    return sample_membrane_signal
+
+
+def generate_two_cell_test_channel_synthetic_data(size_img=(1024, 1024), cell_radius=10, nuc_radius=3, memb_diameter=5,
+                                                  nuc_cell_ids=None, memb_cell_ids=None):
+    """
+    This function generates the complete package of channel-level synthetic data we're looking for
+
+    Args:
+        size_img (tuple): the dimensions of the image we wish to generate
+        cell_radius (int): the radius of each cell
+        nuc_radius (int): the radius of each nucleus
+        memb_diameter (int): the diameter of each membrane
+        nuc_cell_ids (list): a list of which cells we wish to generate nuclear signal for, if None assume just cell 1
+        memb_cell_ids (list): a list of which cells we wish to generate membrane signal for, if None assume just cell 2
+
+    Returns:
+        sample_segmentation_mask (numpy): an array which contains the labeled cell regions
+        sample_nuclear_signal (TIF): defines the nuclear signal for the desired cells
+        sample_membrane_signal (TIF): defines the membrane signal for the desired cells
```

Comment:
(TIF) is a little vague since it could be anything from a string, file pointer, numpy ndarray, mibitiff object, etc.  I'd change it to (numpy).

Reply:
Yeah, meant to make that numpy. No idea why I put TIF there.